src/Reflection/Annotations/AnnotationsMethodsClassReflectionExtension.php

```diff
@@ -59,20 +59,21 @@ private function createMethods(ClassReflection $classReflection): array
 
 		$typeMap = $this->fileTypeMapper->getTypeMap($fileName);
 
-		preg_match_all('#@method\s+(?:(.*)\s+)?([a-zA-Z0-9_]+)\(.*\)#', $docComment, $matches, PREG_SET_ORDER);
+		preg_match_all('#@method\s+(?:(?P<IsStatic>static)\s+)?(?:(?P<Type>[^\s]*)\s+)?(?P<MethodName>[a-zA-Z0-9_]+)\(.*\)#', $docComment, $matches, PREG_SET_ORDER);
 		foreach ($matches as $match) {
-			$typeStringCandidate = $match[1];
-			if (preg_match('#' . FileTypeMapper::TYPE_PATTERN . '#', $typeStringCandidate, $typeStringMatches)) {
-				$typeString = $typeStringMatches[1];
+			$isStatic = !empty($match['IsStatic']);
```

Comment:
Please don't use `empty` but something more specific like `$match['IsStatic'] === 'static`. `empty` can hide a lot of problems.